transports/msmq/delayed-delivery_content_msmqtransport_[2,).partial.md

```diff
@@ -0,0 +1,21 @@
+
+To configure MSMQ to enable delayed message delivery use the following API:
+
+snippet: delayed-delivery
+
+The code above sets up a SQL Server delayed delivery store. It is the only store that ships with the transport. By default it uses a table named after then endpoint with `.delayed` suffix but that name can be customized. 
+
+When using a different database users can provide their own implementations of `IDelayedMessageStore` interface. The delayed delivery handling mechanism has two parts. First, a delayed message is stored using the `Store` method. 
+
+The second part is polling the store for due delayed messages. This is accomplished by periodically calling `FetchNextDueTimeout`. As long as this method returns a message, the mechanism continues to pull more messages. If there are no more due messages, `Next` is invoked to determine how much time is left until the next due timeout. The polling mechanism pauses until that time. If another delayed message is persisted in the meantime, the `Store` method wakes up the polling thread.
+
+When a due delayed message is found, it is forwarded to the destination and then removed from the store using the `Remove` method. In case of an unexpected exception occurs during forwarding the failure is registered using `IncrementFailureCount`. If configured number of retries is exhaused the message is forwarded to the configured `error` queue. Refer to API documentation for details regarding the functionality expected in each method.
+
+### Consistency
+
+In the `TransactionScope` [transaction mode](/transports/transactions.md) the `FetchNextDueTimeout` and `Remove` operations are exeucted within a single `TransactionScope` with forwarding the message to the destination queue. The delayed delivery store is expected to enlist in a distributed transaction (the built-in SQL Server store supports this mode of operation). 
+
+In lower transaction modes the `FetchNextDueTimeout` and `Remove` operations are executed in the same `TransactionScope` but the forwarding is executed in a separate scope. This means that in case of failure in `Remove`, the message can be sent to the destination multiple times. The destination endpoint has to handle the duplicates, either via the [Outbox](/nservicebus/outbox/) or custom deduplication mechanism.
+
+The built-in SQL Server store uses locks the delayed message row in the `FetchNextDueTimeout` operation to prevent multiple instances of the same endpoint from attempting to deliver the same due delayed message. Custom implementation of the store are also expected to use some form of a lock mechanism (e.g. Blob Storage lease locks).
```

Comment:
```suggestion
The built-in SQL Server store applies a pessimistic lock on the delayed message row in the `FetchNextDueTimeout` operation to prevent multiple instances of the same endpoint from attempting to deliver the same due delayed message. Custom implementation of the store are also expected to use some form of a lock mechanism (e.g. Blob Storage lease locks).
```